tests/src/test/java/alluxio/underfs/sleepfs/SleepingUnderFileSystem.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.sleepfs;
+
+import alluxio.AlluxioURI;
+import alluxio.underfs.UnderFileStatus;
+import alluxio.underfs.local.LocalUnderFileSystem;
+import alluxio.underfs.options.CreateOptions;
+import alluxio.underfs.options.DeleteOptions;
+import alluxio.underfs.options.FileLocationOptions;
+import alluxio.underfs.options.MkdirsOptions;
+import alluxio.underfs.options.OpenOptions;
+import alluxio.util.CommonUtils;
+import alluxio.util.io.PathUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+
+/**
+ * An under file system for testing that sleeps a predefined amount of time before executing an
+ * operation. The operation behavior will be equivalent to that of {@link LocalUnderFileSystem}.
+ */
+public class SleepingUnderFileSystem extends LocalUnderFileSystem {
+  private final SleepingUnderFileSystemOptions mOptions;
+
+  /**
+   * Creates a new SleepingUnderFileSystem for the given uri.
```

Comment:
Generally the IDE will give the option to update in comments and javadoc as well. I'm fine with updating this for the sake of consistency.